### PR TITLE
Add initial support for ibis.random function

### LINKED ISSUE #1450: Support for RANDOM-like function
The various backend spell this function slightly differently.

### PR DESCRIPTION
**Pull request by jrbourbeau**
This PR adds an `ibis.random` function which returns a random number between 0 (inclusive) and 1 (exclusive). I've started with initial support for the postgresql and mysql backends

Closes #1450

### CODE REVIEW

File: ibis/expr/api.py [hunk 1]
```diff
@@ -1344,6 +1345,11 @@ def clip(arg, lower=None, upper=None):
     return op.to_expr()
 
 
+def random():
```
Comment (jreback) [inline ibis/expr/api.py]: let's create a ibis.expr.random; the goal is to produce a random scalar yes? reference the documentation https://docs.python.org/3/library/random.html, which i assume is the goal here.

File: ibis/expr/operations.py [hunk 2]
```diff
@@ -2450,6 +2450,11 @@ def output_type(self):
         return dt.timestamp.scalar_type()
 
 
+class Random(Constant):
```
Comment (jreback) [inline ibis/expr/operations.py]: this could be RandomScalar. you could eventually expand this to a column op as well.

Comment (xmnlab) [reply ibis/expr/operations.py]: @jrbourbeau do you still have any interesting on moving this forward? let me know if you need any help! thanks for having been working on that.

Comment (jrbourbeau) [reply ibis/expr/operations.py]: Apologies for the delayed response. I've made updates in response to @jreback's initial feedback, please let me know if there's anything else you'd like me to include in this PR

**Comment by jrbourbeau** [comment]
The `LinuxBuildConda` CI failure seems unrelated to the changes here. Can I get a restart for that build?

**Comment by jrbourbeau** [comment]
I see, thanks for info!

**Comment by xmnlab** [comment]
@jrbourbeau also would be good to have this new operation into the documentation. 
maybe the best place to add that would be https://docs.ibis-project.org/api.html#top-level-expression-apis

thanks for working on that!

**Comment by xmnlab** [comment]
CI windows issues probably related to https://github.com/conda-forge/python-feedstock/pull/325

**Comment by jrbourbeau** [comment]
Good to know, thanks for investigating. It's been a few hours since the fix was merged, would you mind restarting the CI build here to see if things are fixed?

**Comment by jrbourbeau** [comment]
Sweet, thanks for the restart. Looks like the Windows CI failures have been resolved

**Comment by xmnlab** [comment]
LinuxBuildConda should be fixed now. @jrbourbeau could you rebase your code pls? thanks!

**Comment by jrbourbeau** [comment]
Thanks for the heads up @xmnlab! Just updated this PR

**Comment by jreback** [comment]
cc @toryhaavik @icexelloss if you have comments.

**Comment by jrbourbeau** [comment]
Thanks for taking a look @jreback. Merged `master` and CI is green

**Comment by jreback** [comment]
thanks @jrbourbeau 

a followup could be to add this to the pandas back end (easy)

**Pull request merged**